tests/test_config_section.py

```diff
@@ -0,0 +1,546 @@
+        Test validation of non-existing key 
+        """
+        result = self.global_config._validate_str_to_bool('does_not_exist')
+        self.assertIsNotNone(result)
+        self.assertIsInstance(result, tuple)
+
+    def test_validate_string(self):
+        """
+        Test validation of correct string value 
+        """
+        result = self.global_config._validate_non_empty_string('log_file')
+        self.assertIsNone(result)
+
+    def test_validate_interval(self):
+        """
+        Test validation of time interval
+        """
+        result = self.global_config._validate_interval()
+        self.assertIsNone(result)
+
+    def test_validate_wrong_interval(self):
+        """
+        Test validation of wrong time interval
+        """
+        self.global_config['interval'] = '10'
+        result = self.global_config._validate_interval()
+        self.assertIsNotNone(result)
+        interval = self.global_config.get('interval')
+        self.assertEqual(interval, MinimumSendInterval)
+
+    def test_validate_missing_interval(self):
+        """
+        Test validation of wrong time interval
+        """
+        self.global_config = GlobalSection('global', None)
+        for key, value in GLOBAL_SECTION_VALUES.items():
+            if key != 'interval':
+                self.global_config[key] = value
+        result = self.global_config._validate_interval()
+        self.assertIsNone(result)
+        interval = self.global_config['interval']
+        self.assertEqual(interval, DefaultInterval)
+
+    def test_validate_configs(self):
+        """
+        Test validation of configs (list of paths to config files)
+        """
+        result = self.global_config._validate_configs()
+        self.assertIsNone(result)
+
+    def test_validate_section_values(self):
+        """
+        Test validation of all config values 
+        """
+        validate_messages = self.global_config.validate()
+        print(validate_messages)
+        expected_results = [
+            ('warning', 'Ignoring unknown configuration option "wrong_bool_value"'),
+            ('warning', 'log_per_config must be a valid boolean, using default. See man virt-who-config for more info'),
+            ('warning', 'Value for reporter_id not set, using default'),
+            ('warning', 'Value for log_dir not set, using default'),
+            ('warning', 'background must be a valid boolean, using default. See man virt-who-config for more info')
+        ]
+        self.assertGreater(len(validate_messages), 0)
+        # self.assertEqual(expected_results, validate_messages)
+
+    def test_validate_default_value(self):
+        """
+        Test the validate method will set default value, when an option
+        is missing and default value exist in DEFAULT
+        """
+        self.global_config.validate()
+        default_value = self.global_config.get('log_dir')
+        self.assertEqual(default_value, DEFAULT_LOG_DIR)
+
+
+MY_SECTION_VALUES = {
```

Comment:
@kahowell Do you mean move `MyConfigSection` to `stubs.py` and split `test_config_section.py` to three files: `test_config_section.py`, `test_global_config_section.py` and `test_virt_config_section.py`?